fdbserver/workloads/BackupCorrectness.actor.cpp

```diff
@@ -656,6 +694,22 @@ struct BackupAndRestoreCorrectnessWorkload : TestWorkload {
 				state std::vector<Standalone<StringRef>> restoreTags;
 				state bool multipleRangesInOneTag = false;
 				state int restoreIndex = 0;
+				// make sure system keys are not present in the restoreRanges as they will get restored first separately
+				// from the rest
+				Standalone<VectorRef<KeyRangeRef>> modifiedRestoreRanges;
+				bool hasSystemRanges = false;
+				for (int i = 0; i < self->restoreRanges.size(); ++i) {
+					if (!SERVER_KNOBS->ENABLE_ENCRYPTION ||
+					    !self->restoreRanges[i].intersects(getSystemBackupRanges())) {
+						modifiedRestoreRanges.push_back_deep(modifiedRestoreRanges.arena(), self->restoreRanges[i]);
+					} else {
+						hasSystemRanges = true;
+					}
+				}
+				self->restoreRanges = modifiedRestoreRanges;
+				if (hasSystemRanges) {
+					wait(clearAndRestoreSystemKeys(cx, self, &backupAgent, targetVersion, lastBackupContainer));
```

Comment:
Inline comment to say this works because we restore the whole cluster instead of partial key range for the test?